lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -283,7 +283,7 @@ public function getListTableForeignKeysSQL($table, $database = null)
     {
         return 'SELECT quote_ident(r.conname) as conname, pg_catalog.pg_get_constraintdef(r.oid, true) as condef
                   FROM pg_catalog.pg_constraint r
-                  WHERE r.conrelid =
+                  WHERE r.conrelid IN
```

Comment:
Isn't it because it ignores the `$database` parameter? Based on the scenario described in https://github.com/doctrine/dbal/pull/3422#issuecomment-463331428, I'd expect the query to select the rows corresponding to the table in its schema, not other tables with the same name in other schemas.

Reply:
The `$database` parameter would not be useful for the query in this function as it would just be a string of the database name. We'd need the name of the schema to be able to make this query be more specific, which this function isn't equipped to receive. 

IMO this is the best solution to this issue. Otherwise you'll have to go down the rabbit trail of how to pass a schema name down to this function. And I don't believe that would be worth the effort (my effort anyway) as this is such a rare situation where there would be two schemas with the same table name AND have foreign key constraints.

Reply:
> The `$database` parameter would not be useful for the query in this function as it would just be a string of the database name. We'd need the name of the schema to be able to make this query be more specific […]

This is what I'm talking about. If I'm not mistaken, `$database` in the DBAL terminology corresponds to the PostgreSQL schema.

Reply:
> If I'm not mistaken, $database in the DBAL terminology corresponds to the PostgreSQL schema.

This is incorrect. `$database` contains the name of the database, not the name of the schema. In PostgreSQL, a database can have multiple schemas: `database -> schemas -> tables` . It's the schema name that is needed for added specificity, not the database name.

So, the `$database` parameter is actually completely useless for this function. I'm not sure why it's even there. My guess is it's just carry-over (copy/paste) from other drivers that override this abstract function (see MySqlPlatform->getListTableForeignKeysSQL()).

Reply:
You're right. DBAL doesn't use `$database` to indicate a schema for PostgreSQL. However, it still doesn't look right to return foreign keys for tables with the same name from all schemas as a workaround. Please see if you can use `schema.table` as the table name since it seems to be supported in:

https://github.com/doctrine/dbal/blob/0a55ce69a634009624f30f11971770ffbeb42099/lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php#L366-L371

Reply:
You've pretty much just shown that the solution you are looking for is already in the `getListTableForeignKeysSQL` function.

`getListTableForeignKeysSQL` utilizes the `getTableWhereClause` function to filter for schema. 
https://github.com/doctrine/dbal/blob/0a55ce69a634009624f30f11971770ffbeb42099/lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php#L290

But the `$table` parameter must have the schema in it (i.e. `$table = 'some_schema.some_table'`), otherwise the `getTableWhereClause` will extrapolate a list of schemas from the users search_path. 

So, passing specificity to a single schema is already possible in this function. The developer can included the schema with the table name in the `$table` parameter ('some_schema.some_table'). But if they don't, it'll fall back on a list of schemas in the DB users search_path.

Therefore, the very simple fix this PR (literally just one key stroke on one line code) is still necessary. And there is no need to figure out how to pass a specific schema to this function - as you have found that it is already available in the code.

With all due respect, I implore you to stop beating around the bush with this question of schema specificity, and please let this PR be merged.

Thank you.